flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -566,6 +562,22 @@ private ClusterSpecification validateClusterResources(
 
 	}
 
+	private void logIfComponentMemNotIntegerMultipleOfYarnMinAllocation(
+			String componentName,
+			int componentMemoryMB,
+			int yarnMinAllocationMB) {
+		int normalizedMemMB = (componentMemoryMB + (yarnMinAllocationMB - 1)) / yarnMinAllocationMB * yarnMinAllocationMB;
```

Comment:
What if `yarnMinAllocationMB` equals to zero, which is the default value set in https://github.com/apache/flink/blob/03b82f9aa81b828b06c61f7f30f92a331ec5120b/flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java#L511

Reply:
Thanks for your comments. I think `yarnMinAllocationMB` will always be greater than 0(the default is 1024m). Otherwise, the Yarn resourcemanager could not start successfully.

You could verify this via adding or removing the `yarn.scheduler.minimum-allocation-mb` in your `yarn-site.xml`.

Reply:
Actually I'm here because I met the problem in our environment. I'm not an expert on YARN but I think we should not rely on the yarn's limitations here. Besides, if our assumption is that `yarnMinAllocationMB` should always be greater than 0, why do we set the default value as 0, which is obviously a wrong value and certainly crashes the job.

Reply:
Yes, i think it is a wrong a default value. We should use `YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_MB` instead.

So in your environment, the `yarn.scheduler.minimum-allocation-mb` is configured to 0 explicitly? Otherwise, i am curious how it could happen.

Reply:
@wangyang0918 Yes. It's explicitely set to 0. And I don't know why...

Reply:
@wangyang0918 BTW, I can fix this today if you don't mind.

Reply:
Even i still believe that it should not be configured to 0 from Yarn side. However, i think Flink could do more things to avoid this.

Feel free to open a ticket. We could make the following two minor changes.
* Make the default value to `YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_MB`, not 0
* Add pre-check for `yarnMinAllocationMB`, it should be greater than 0.

Reply:
In your case, you still need to configure `yarn.scheduler.minimum-allocation-mb` to a positive number(i.e. greater than 0). It is a mandatory requirement of Yarn.

Reply:
@wangyang0918 Yes. The change is not for my case, just make the default value here more sensible.